interface-spec.html

```diff
@@ -334,8 +354,20 @@ <h3><dfn>Quad</dfn> interface</h3>
       <code>NamedNode</code>, <code>BlankNode</code> or <code>Variable</code>.
     </p>
     <p>
-      <dfn>equals()</dfn> returns <code>true</code> if and only if the argument is a) of the same
-      type b) has all components equal.
+      <dfn id="quad-equals">equals()</dfn> returns <code>true</code>
+      when called with parameter <code>other</code>
+      on an object <code>quad</code> if and only if
+      all of the conditions below hold:
+    </p>
+    <ul>
+      <li><code>other</code> is <em>neither</em> <code>null</code> nor <code>undefined</code>;</li>
+      <li><code>quad.subject.equals(other.subject)</code> evaluates to a truthy value;</li>
+      <li><code>quad.predicate.equals(other.predicate)</code> evaluates to a truthy value;</li>
+      <li><code>quad.object.equals(other.object)</code> evaluates to a truthy value;</li>
+      <li><code>quad.graph.equals(other.graph)</code> evaluates to a truthy value;</li>
```

Comment:
The idea was to relax, so you don’t have to write `=== true`; you might not own the implementations of the component methods (terms could be imported).

Reply:
Okay, so then we also have to check that `quad.[subject/predicate/object/graph].equals` are indeed functions before using them yes? Otherwise they will throw an error.